Update the Kafka recovery procedure with KRaft notes (#10728)
strimzi/strimzi-kafka-operator: documentation/modules/managing/proc-cluster-recovery-volume.adoc

    *apiVersion: v1*
    *kind: PersistentVolumeClaim*
    *name: data-0-my-cluster-kafka-2*

>     *namespace: my-project*

Reviewer: Should we stick with `myproject` everywhere? That is what we use int he installation files etc.

Author: There was a mix of the two, but I can move all to myproject.

Reviewer: Thanks

.. Apply the `KafkaUser` resource configuration:
+
[source,shell]
kubectl apply -f <user_configuration_file> -n my-project

. Deploy the Kafka cluster using the original configuration for the `Kafka` resource.

> If it's a KRaft cluster, add the annotation `strimzi.io/pause-reconciliation="true"` to the YAML file. 

Reviewer 2: This should be a KRaft step like the unpause.

Author: Yeah, let me split the procedures.

+
[source,shell]
----
kubectl apply -f <kafka_resource_configuration>.yaml -n my-project
----

> .. KRaft only: recover the `clusterId` value from one of the volumes by spinning up a temporary pod:

Reviewer: Maybe you could say something like `If you don't know the cluster ID of the cluster (e.g. from logs or copies of the Kafka custom resource)`.

+
[source,shell]
----
PVC_NAME="data-0-my-cluster-kafka-0"
COMMAND="grep cluster.id /disk/kafka-log*/meta.properties | awk -F'=' '{print \$2}'"
kubectl run tmp -itq --rm --restart "Never" --image "foo" --overrides "{\"spec\":
  {\"containers\":[{\"name\":\"busybox\",\"image\":\"busybox\",\"command\":[\"/bin/sh\",
  \"-c\",\"$COMMAND\"],\"volumeMounts\":[{\"name\":\"disk\",\"mountPath\":\"/disk\"}]}],
  \"volumes\":[{\"name\":\"disk\",\"persistentVolumeClaim\":{\"claimName\":
  \"$PVC_NAME\"}}]}}" -n my-project

> ----

Author: Well, yes, there could be. This is just one method to retrieve the clusterId by spinning up a pod providing the template. This whole procedure is for expert users, so I guess they would know how to customize the template if needed.

> .. KRaft only: edit the Kafka resource using the `--subresource status` flag and set the `clusterId`:

Reviewer 2: I removed the subresource status bit here. If we want to include it (as the best way to do this), maybe we should be more explicit about the steps?

Return the cluster ID from the status of the `Kafka` resource: ` kubectl get kafka <kafka_resource_name> -o jsonpath='{.status}' | jq`  
Update the cluster ID: ...

> .. KRaft only: edit the Kafka resource using the `--subresource status` flag and set the `clusterId`:

Reviewer 2: Does this justify any warning about making sure the ID is correct?

> .. KRaft only: edit the Kafka resource using the `--subresource status` flag and set the `clusterId`:

Author: The `--subresource status` is absolutely required.

> Does this justify any warning about making sure the ID is correct?

In the previous point, we say you need to retrieve it from previous deployment in one way or the other, so I think it is clear that it must be correct.

> .. KRaft only: edit the Kafka resource using the `--subresource status` flag and set the `clusterId`:

Reviewer 2: Sorry, I wasn't clear.  It was the process of editing it, I was wondering about. I thought you were implying using kubectl commands to edit it, rather just a straightforward edit to the `status` property.

> .. KRaft only: edit the Kafka resource using the `--subresource status` flag and set the `clusterId`:

Author: Do you want to add the edit command? We could also use the patch command, but I see that this is never used across the doc.

> .. KRaft only: edit the Kafka resource using the `--subresource status` flag and set the `clusterId`:

Reviewer: I think having the full edit command there might be useful ... `kubectl edit kafka <cluster-name> --subresource status`

+
[source,shell]
----
status:
  clusterId: w2MRy6xsQEqqWQkhLvbTYg
----

> .. KRaft only: unpause the Kafka resource reconciliation:

Reviewer 2: ```suggestion
.. KRaft only: Unpause the Kafka resource reconciliation:
```

.. Apply the `KafkaUser` resource configuration:
+
[source,shell]
kubectl apply -f <user_configuration_file> -n myproject

> . Deploy the Kafka cluster using the original configuration for the `Kafka` resource.
> It's important to add the annotation `strimzi.io/pause-reconciliation="true"` to the YAML file.

Reviewer 2: ```suggestion
Add the annotation `strimzi.io/pause-reconciliation="true"` to the original configuration for the `Kafka` resource, and then deploy the Kafka cluster using the updated configuration.
```

+
[source,shell]
----
kubectl apply -f <kafka_resource_configuration>.yaml -n myproject
----

> .. Recover the original `clusterId` from logs or copies of the `Kafka` custom resource.

Reviewer 2: I think these sub-steps can just be steps now?
